mux_test.go

```diff
@@ -207,3 +207,36 @@ func newRequest(method, urlStr string, body io.Reader) *http.Request {
 	}
 	return req
 }
+
+func TestMethodMatch(t *testing.T) {
+	p := New()
+	p.Patch("/foo/bar", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+
+	// Test to see if we get a 405 Method Not Allowed errors from trying to
+	// issue a GET request to a handler that only supports the PATCH method.
+	req, err := http.NewRequest("GET", "/foo/bar", nil)
```

Comment:
Since there's a new helper, here and below you can just write `req := newRequest("GET", "/foo/bar", nil)`